### PR TITLE
Add support for macOS Rapid Security Response updates

### PR DESCRIPTION
**Pull request by nstrauss**
## Description
Apple is adding a new update type to macOS called Rapid Security Responses. These updates change the build version, but not the OS version reported by `sw_vers`. Instead, a new `ProductVersionExtra` key is returned to denote a specific supplemental version only when a RSR is installed. If no RSR is installed the key is not present. 

```
$ sw_vers
ProductName: macOS
ProductVersion: 13.3
ProductVersionExtra: (a)
BuildVersion: 22E7752300f
```

https://support.apple.com/guide/deployment/rapid-security-responses-dep93ff7ea78/web

This PR adds support for reporting on the RSR version when a patch is installed. The information is important to Mac admins since it gives a full picture of current OS build, whether a security vulnerability has been patched, and, depending on Apple's future plans, if there is functionality change which could impact Chef. I'll admit the spec test isn't totally useful, especially since most machines running tests likely won't have an RSR, but it does match real world scenarios. 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Chore (non-breaking change that does not add functionality or fix an issue)

## Checklist:

- [x] I have read the **CONTRIBUTING** document.
- [x] I have run the pre-merge tests locally and they pass.
- [x] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [x] All commits have been signed-off for [the Developer Certificate of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

### CODE REVIEW

**Review by tpowell-progress — APPROVED** [review]
Confirmed tests pass.

**Review by marcparadise — DISMISSED** [review]
Please update the commit message to match the PR wording so we have local/repo access to this history and background.

File: spec/unit/plugins/darwin/platform_spec.rb [hunk 1]
```diff
@@ -41,6 +41,11 @@
     expect(@plugin[:platform_version]).to eq("10.15.6")
   end
 
+  it "sets platform_version_extra to ProductVersionExtra" do
+    @plugin.run
+    expect(@plugin[:platform_version_extra]).to eq("(a)").or be_nil
```
Comment (jaymzh) [inline spec/unit/plugins/darwin/platform_spec.rb:46]: this isn't a meaningful test - it'll be null given the test data... but it'd also be null if you were parsing wrong and the code was broken.

You can have this test be "returns null if no PVE is available", but then add another test that changes the test data (currently set on line 25) and ensure it's parsed properly.

Thanks!

**Review by jaymzh — CHANGES_REQUESTED** [review]

File: spec/unit/plugins/darwin/platform_spec.rb (hunk 1 above)
Comment (nstrauss) [reply spec/unit/plugins/darwin/platform_spec.rb]: @jaymzh Thanks for the guidance here. I'm not overly familiar with writing tests, but did what I could here. I've updated `sw_vers` to match more recent output on macOS as well. Please let me know if there's anything else I should do to get the PR approved.

**Review by jaymzh — APPROVED** [review]
We theoretically should move `@stdout` to be a `let` instead of a classvar, but we won't hold you to that standard since you found it this way.

Tests look good, code looks good. I'll merge if the tests pass.

**Comment by jaymzh** [comment]
Thanks!

**Pull request merged**